api/capi/src/nnstreamer-capi-util.c

```diff
@@ -933,46 +935,35 @@ ml_validate_model_file (char **model, unsigned int num_models,
 
   switch (*nnfw) {
     case ML_NNFW_TYPE_ANY:
-      /** @todo Make sure num_models is correct */
-      if (g_str_has_suffix (path_down, ".tflite")) {
-        /**
-         * .tflite is supported by both tensorflow and nnfw.
-         * Priority decided with ini file.
-         */
-        gboolean nnfw_runtime_priority =
-            nnsconf_get_custom_value_bool ("nnfw-runtime",
-            "prioritize_tflite_ext", FALSE);
-        gboolean available_nnfw =
-            ml_nnfw_is_available (ML_NNFW_TYPE_NNFW, ML_NNFW_HW_ANY);
-        gboolean available_tflite =
-            ml_nnfw_is_available (ML_NNFW_TYPE_TENSORFLOW_LITE, ML_NNFW_HW_ANY);
-
-        if ((nnfw_runtime_priority && available_nnfw) ||
-            (!nnfw_runtime_priority && !available_tflite)) {
-          ml_logi ("The given model [%s] is supposed a nnfw model.", model[0]);
-          *nnfw = ML_NNFW_TYPE_NNFW;
-        } else {
-          ml_logi ("The given model [%s] is supposed a tensorflow-lite model.",
-              model[0]);
-          *nnfw = ML_NNFW_TYPE_TENSORFLOW_LITE;
-        }
-      } else if ((g_str_has_suffix (path_down, ".so") &&
-              g_str_has_suffix (path_down_2, ".nb")) ||
-          (g_str_has_suffix (path_down, ".nb") &&
-              g_str_has_suffix (path_down_2, ".so"))) {
+      detected_fw =
+          gst_tensor_filter_framework_auto_detection ((const gchar **) model,
+          num_models);
+
+      if (detected_fw == NULL) {
+        ml_loge ("The given model [%s] has unknown or not supported extension.",
+            model[0]);
+        status = ML_ERROR_INVALID_PARAMETER;
+      } else if (g_ascii_strcasecmp (detected_fw, "nnfw") == 0) {
+        ml_logi ("The given model [%s] is supposed a nnfw model.", model[0]);
+        *nnfw = ML_NNFW_TYPE_NNFW;
+      } else if (g_ascii_strcasecmp (detected_fw, "tensorflow-lite") == 0) {
+        ml_logi ("The given model [%s] is supposed a tensorflow-lite model.",
+            model[0]);
+        *nnfw = ML_NNFW_TYPE_TENSORFLOW_LITE;
+      } else if (g_ascii_strcasecmp (detected_fw, "vivante") == 0) {
```

Comment:
I think you missed one line,
*nnfw = ML_NNFW_TYPE_VIVANTE;

Reply:
Oh.. Thank you. Updated